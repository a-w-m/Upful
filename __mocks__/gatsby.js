const React = require("react")
const gatsby = jest.requireActual("gatsby")

const mockDate = new Date("2019-04-07T10:20:30Z")

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(
    // these props are invalid for an `a` tag
    ({
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      to,
      ...rest
    }) =>
      React.createElement("a", {
        ...rest,
        href: to,
      })
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn().mockReturnValue({
        "allMarkdownRemark": {
          "edges": [
            {
              "node": {
                "fields": {
                  "slug": "/about/"
                },
                "frontmatter": {
                  "title": "About"
                }
              }
            },
            {
              "node": {
                "fields": {
                  "slug": "/returns/"
                },
                "frontmatter": {
                  "title": "Returns"
                }
              }
            }
          ]
        },
        "site": {
          "siteMetadata": {
            "menuLinks": {
              "categories": [
                {
                  "name": "women",
                  "slug": "/women/"
                },
                {
                  "name": "kids",
                  "slug": "/kids/"
                }
              ]
            }
          }
        },
        "file": {
          "id": "f1fc4c65-65c9-5962-b6a8-9a265bb2e617",
          "childImageSharp": {
            "gatsbyImageData": {
              "layout": "constrained",
              "backgroundColor": "#f8f8f8",
              "images": {
                "fallback": {
                  "src": "/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png",
                  "srcSet": "/static/55955cd5db9ac5b9496eb4fb43c87585/6a639/upful-gold-frame-logo.png 63w,\n/static/55955cd5db9ac5b9496eb4fb43c87585/2fd20/upful-gold-frame-logo.png 125w,\n/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png 250w,\n/static/55955cd5db9ac5b9496eb4fb43c87585/82c11/upful-gold-frame-logo.png 500w",
                  "sizes": "(min-width: 250px) 250px, 100vw"
                },
                "sources": [
                  {
                    "srcSet": "/static/55955cd5db9ac5b9496eb4fb43c87585/74c72/upful-gold-frame-logo.webp 63w,\n/static/55955cd5db9ac5b9496eb4fb43c87585/d66e1/upful-gold-frame-logo.webp 125w,\n/static/55955cd5db9ac5b9496eb4fb43c87585/e7160/upful-gold-frame-logo.webp 250w,\n/static/55955cd5db9ac5b9496eb4fb43c87585/5f169/upful-gold-frame-logo.webp 500w",
                    "type": "image/webp",
                    "sizes": "(min-width: 250px) 250px, 100vw"
                  }
                ]
              },
              "width": 250,
              "height": 250
            }
          }
        }
  }),
  Date: jest.spyOn(global.Date, "now").mockImplementation(() => mockDate),
}
