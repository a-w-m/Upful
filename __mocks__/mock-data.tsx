export const pageQueryData = {
  data: {
    markdownRemark: {
      frontmatter: {
        customField1: {
          name: "size",
          values: [
            {
              name: "small",
              priceChange: 0,
            },
            {
              name: "medium",
              priceChange: 0,
            },
            {
              name: "large",
              priceChange: 0,
            },
          ],
        },
        customField2: {
          name: "color",
          values: [
            {
              name: "blue",
              priceChange: 0,
            },
            {
              name: "grey",
              priceChange: 0,
            },
            {
              name: "black",
              priceChange: 0,
            },
          ],
        },
        title: "Pietsie Atlin Blouse",
        price: 240,
        date: "2020-12-29T00:00:00.000Z",
        description:
          "Pietsie Campbell creates all of her garments out of deadstock, vintage fabrics, and natural fibers. Each collection is made in sunny California and embodies the effortlessly chic look we all strive for. Our favorite loose, flowing blouse in a cotton and linen blend. Slightly fitted across the shoulders. Wear tucked or loose, with the sleeves rolled up or down. Delicate pleating at shoulders, bell sleeves, pleated cuffs. Trapeze shape, easy fit. Pulls over, vintage shell buttons on a placket that ends mid way. Delicate hand-pleated detail on chest, stand-up prairie collar.",
        id: "pietsie-atlin-blouse",
        image: {
          childImageSharp: {
            fluid: {
              base64:
                "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAQEBAQAAAAAAAAAAAAAAAAQDBQL/xAAVAQEBAAAAAAAAAAAAAAAAAAACAf/aAAwDAQACEAMQAAABo5/qM3ZaNxlHOsAv/8QAGxAAAgMAAwAAAAAAAAAAAAAAAhEAAQMQMjP/2gAIAQEAAQUCNAZG+Nms1Z1Ow5etKf/EABcRAQADAAAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8BYmv/xAAVEQEBAAAAAAAAAAAAAAAAAAAQAf/aAAgBAgEBPwEh/8QAGRAAAgMBAAAAAAAAAAAAAAAAAAEQITER/9oACAEBAAY/AmU6WQxdixGH/8QAHBABAAMAAgMAAAAAAAAAAAAAAQARITFRQXGB/9oACAEBAAE/IXc+4GAexFvZzo0GrKj8QZgMPIs6gqnJhwyf/9oADAMBAAIAAwAAABBg6AP/xAAYEQACAwAAAAAAAAAAAAAAAAABEBExQf/aAAgBAwEBPxCjCYX/xAAZEQACAwEAAAAAAAAAAAAAAAAAARARITH/2gAIAQIBAT8QWqzThx//xAAaEAEAAwEBAQAAAAAAAAAAAAABABEhMUFR/9oACAEBAAE/EHjoQXw9Zt3EBTGMhU2ZF4YEW35sIxq0nx8nAaNWAyzXA74cHfsrTdJZ2Gw7kOz/2Q==",
              aspectRatio: 1,
              src:
                "/static/184dc14fc5e96a12c4f7c6ee0f71f32a/14b42/pietsie-atlin-blouse.jpg",
              srcSet:
                "/static/184dc14fc5e96a12c4f7c6ee0f71f32a/f836f/pietsie-atlin-blouse.jpg 200w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/2244e/pietsie-atlin-blouse.jpg 400w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/14b42/pietsie-atlin-blouse.jpg 800w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/16310/pietsie-atlin-blouse.jpg 1024w",
              sizes: "(max-width: 800px) 100vw, 800px",
            },
          },
        },
      },
      fields: {
        slug: "/pietsie-atlin-blouse/",
      },
    },
  },
}

export const pageQueryDataNullCustomFields = {
  data: {
    markdownRemark: {
      frontmatter: {
        customField1: null,
        customField2: null,
        title: "Pietsie Atlin Blouse",
        price: 240,
        date: "2020-12-29T00:00:00.000Z",
        description:
          "Pietsie Campbell creates all of her garments out of deadstock, vintage fabrics, and natural fibers. Each collection is made in sunny California and embodies the effortlessly chic look we all strive for. Our favorite loose, flowing blouse in a cotton and linen blend. Slightly fitted across the shoulders. Wear tucked or loose, with the sleeves rolled up or down. Delicate pleating at shoulders, bell sleeves, pleated cuffs. Trapeze shape, easy fit. Pulls over, vintage shell buttons on a placket that ends mid way. Delicate hand-pleated detail on chest, stand-up prairie collar.",
        id: "pietsie-atlin-blouse",
        image: {
          childImageSharp: {
            fluid: {
              base64:
                "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAQEBAQAAAAAAAAAAAAAAAAQDBQL/xAAVAQEBAAAAAAAAAAAAAAAAAAACAf/aAAwDAQACEAMQAAABo5/qM3ZaNxlHOsAv/8QAGxAAAgMAAwAAAAAAAAAAAAAAAhEAAQMQMjP/2gAIAQEAAQUCNAZG+Nms1Z1Ow5etKf/EABcRAQADAAAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8BYmv/xAAVEQEBAAAAAAAAAAAAAAAAAAAQAf/aAAgBAgEBPwEh/8QAGRAAAgMBAAAAAAAAAAAAAAAAAAEQITER/9oACAEBAAY/AmU6WQxdixGH/8QAHBABAAMAAgMAAAAAAAAAAAAAAQARITFRQXGB/9oACAEBAAE/IXc+4GAexFvZzo0GrKj8QZgMPIs6gqnJhwyf/9oADAMBAAIAAwAAABBg6AP/xAAYEQACAwAAAAAAAAAAAAAAAAABEBExQf/aAAgBAwEBPxCjCYX/xAAZEQACAwEAAAAAAAAAAAAAAAAAARARITH/2gAIAQIBAT8QWqzThx//xAAaEAEAAwEBAQAAAAAAAAAAAAABABEhMUFR/9oACAEBAAE/EHjoQXw9Zt3EBTGMhU2ZF4YEW35sIxq0nx8nAaNWAyzXA74cHfsrTdJZ2Gw7kOz/2Q==",
              aspectRatio: 1,
              src:
                "/static/184dc14fc5e96a12c4f7c6ee0f71f32a/14b42/pietsie-atlin-blouse.jpg",
              srcSet:
                "/static/184dc14fc5e96a12c4f7c6ee0f71f32a/f836f/pietsie-atlin-blouse.jpg 200w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/2244e/pietsie-atlin-blouse.jpg 400w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/14b42/pietsie-atlin-blouse.jpg 800w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/16310/pietsie-atlin-blouse.jpg 1024w",
              sizes: "(max-width: 800px) 100vw, 800px",
            },
          },
        },
      },
      fields: {
        slug: "/pietsie-atlin-blouse/",
      },
    },
  },
}

export const fluid = {
  base64:
    "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAQEBAQAAAAAAAAAAAAAAAAQDBQL/xAAVAQEBAAAAAAAAAAAAAAAAAAACAf/aAAwDAQACEAMQAAABo5/qM3ZaNxlHOsAv/8QAGxAAAgMAAwAAAAAAAAAAAAAAAhEAAQMQMjP/2gAIAQEAAQUCNAZG+Nms1Z1Ow5etKf/EABcRAQADAAAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8BYmv/xAAVEQEBAAAAAAAAAAAAAAAAAAAQAf/aAAgBAgEBPwEh/8QAGRAAAgMBAAAAAAAAAAAAAAAAAAEQITER/9oACAEBAAY/AmU6WQxdixGH/8QAHBABAAMAAgMAAAAAAAAAAAAAAQARITFRQXGB/9oACAEBAAE/IXc+4GAexFvZzo0GrKj8QZgMPIs6gqnJhwyf/9oADAMBAAIAAwAAABBg6AP/xAAYEQACAwAAAAAAAAAAAAAAAAABEBExQf/aAAgBAwEBPxCjCYX/xAAZEQACAwEAAAAAAAAAAAAAAAAAARARITH/2gAIAQIBAT8QWqzThx//xAAaEAEAAwEBAQAAAAAAAAAAAAABABEhMUFR/9oACAEBAAE/EHjoQXw9Zt3EBTGMhU2ZF4YEW35sIxq0nx8nAaNWAyzXA74cHfsrTdJZ2Gw7kOz/2Q==",
  aspectRatio: 1,
  src:
    "/static/184dc14fc5e96a12c4f7c6ee0f71f32a/14b42/pietsie-atlin-blouse.jpg",
  srcSet:
    "/static/184dc14fc5e96a12c4f7c6ee0f71f32a/f836f/pietsie-atlin-blouse.jpg 200w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/2244e/pietsie-atlin-blouse.jpg 400w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/14b42/pietsie-atlin-blouse.jpg 800w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/16310/pietsie-atlin-blouse.jpg 1024w",
  sizes: "(max-width: 800px) 100vw, 800px",
}


