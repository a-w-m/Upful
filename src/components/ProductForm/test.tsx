import React from "react"
import { cleanup, fireEvent, render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Option from "./index"
import { getProductFormData } from "../../../__mocks__/mock-data"

describe("Option", () => {
  const dispatchMock = jest.fn()

  beforeEach(() => {
    render(
      <Option
        productOptions={getProductFormData()}
        dispatch={dispatchMock}
        selected={null}
      ></Option>
    )
  })

  // clear mocks to prevent userEvent history from persisting across tests
  afterEach(() => {
    cleanup()
    jest.clearAllMocks()
  })

  it("matches snapshot", () => {
    const form = screen.getByRole("form")
    expect(form).toMatchSnapshot()
  })

  it("should display name of custom field", () => {
    const field = screen.getByText(/Size/i)
    expect(field).toBeInTheDocument()
  })

  it("should display each radio input", () => {
    const inputs = screen.getAllByRole("radio")
    expect(inputs).toHaveLength(getProductFormData()[0].options.length)
  })

  it("should display each label", ()=>{
    const labels = document.querySelectorAll('label')
    expect(labels).toHaveLength(getProductFormData()[0].options.length)
  })

  it("should display name of option that user clicks", async () => {
    const user = userEvent.setup()
    const option = screen.getByRole("radio", { name: /large/i })
    await user.click(option)
    const selected = screen.getByText(/Size | Large/)
    expect(selected).toBeInTheDocument()
  })

  it("should call dispatch when user clicks option", async () => {
    const user = userEvent.setup()
    const option = screen.getByRole("radio", { name: /large/i })
    await user.click(option)
    expect(dispatchMock).toHaveBeenCalledTimes(1)
  })
})
