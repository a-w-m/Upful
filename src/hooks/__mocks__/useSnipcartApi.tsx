import { getSnipcartApiData } from "../../utils/test/data"

const useSnipcartApi =  jest.fn().mockImplementation(() => {
    return [getSnipcartApiData().state, getSnipcartApiData().dispatch]
  })

export default useSnipcartApi
