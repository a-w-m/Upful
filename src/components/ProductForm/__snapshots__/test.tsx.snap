// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option matches snapshot 1`] = `
<form
  class="sc-hKMtZM dtFrcj"
  name="Select Size"
>
  <span
    aria-label="selected option"
    class="sc-dkzDqf FRBwX"
  >
    Size | 
  </span>
  <input
    class="sc-bczRLJ YJXLH"
    id="Extra Small"
    name="Size"
    type="radio"
    value="Extra Small"
  />
  <label
    class="sc-gsnTZi biaMNL"
    for="Extra Small"
  >
    Extra Small
  </label>
  <input
    class="sc-bczRLJ YJXLH"
    id="Small"
    name="Size"
    type="radio"
    value="Small"
  />
  <label
    class="sc-gsnTZi biaMNL"
    for="Small"
  >
    Small
  </label>
  <input
    class="sc-bczRLJ YJXLH"
    id="Medium"
    name="Size"
    type="radio"
    value="Medium"
  />
  <label
    class="sc-gsnTZi biaMNL"
    for="Medium"
  >
    Medium
  </label>
  <input
    class="sc-bczRLJ YJXLH"
    id="Large"
    name="Size"
    type="radio"
    value="Large"
  />
  <label
    class="sc-gsnTZi biaMNL"
    for="Large"
  >
    Large
  </label>
</form>
`;
